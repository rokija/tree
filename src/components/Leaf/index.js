import React from 'react'

const Leaf = ({
  onOpenInput, paddingLeft, index, value,
}) => (
  <div style={{ paddingLeft }}>
    {value}
    <button onClick={() => onOpenInput(index)}>+</button>
  </div>
)

export default Leaf
