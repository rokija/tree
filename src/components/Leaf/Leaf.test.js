import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Leaf from '.'

describe('Leaf', () => {
  const wrapper = shallow(<Leaf />)

  it('renders properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
