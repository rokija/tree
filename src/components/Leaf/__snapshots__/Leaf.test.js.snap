// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaf renders properly 1`] = `
<div
  style={
    Object {
      "paddingLeft": undefined,
    }
  }
>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
